homeassistant/components/zeroconf.py

```diff
@@ -42,7 +47,13 @@ def setup(hass, config):
         'requires_api_password': requires_api_password,
     }
 
-    host_ip = util.get_local_ip()
+    host_ip = config.get(CONF_LOCAL_IP)
+    if host_ip is not None:
+        host_ip = str(host_ip)
```

Comment:
We should not do this here, we should do this in the config schema: `vol.All(ip_address, str)`

Reply:
Found the pattern in other components... guess I'll correct them